treelib/tree.py

```diff
@@ -824,6 +824,26 @@ def to_json(self, with_data=False, sort=True, reverse=False):
         """To format the tree in JSON format."""
         return json.dumps(self.to_dict(with_data=with_data, sort=sort, reverse=reverse))
 
+    def from_json(self, json_str):
+        """
+        Load tree from exported JSON string.
+
+        :param json_str: json string that exported by to_json method
+        """
+        def _iter(nodes, parent_id):
+            for k,v in nodes.items():
+                children = v.get('children', None)
+                data = v.get('data', None)
+                if children:
+                    yield (k, data, parent_id)
+                    for child in children:
+                        yield from _iter(child, k)
```

Comment:
Integration test failed, ```yield from ...``` syntax was introduced in python 3.3. To support 2.6 and 2.7 I'd suggest the following change:
```suggestion
                        for i in _iter(child, k):
                            yield i
```